### PR TITLE
Feature/client

### PR DESCRIPTION
**Pull request by l8518**

### CODE REVIEW

File: server/routes/urlshortener.js [hunk 1]
```diff
@@ -1,20 +1,64 @@
...
+        var shortenedUrlId = code.encode(index);
+        const shortBaseUrl = WINDOW_LOCATION_ORIGIN + ':' + PORT;
+        let shortenedUrlObject = new urlShorten(url, shortBaseUrl + '/' + shortenedUrlId, shortenedUrlId);
+        shortenedUrlObject = urlRepository.add(shortenedUrlObject);
+
+        return res.status(201).json(shortenedUrlId);
+    })
+
+    app.delete('/', function (req, res) {
+        urlMap = {};
```
Comment (luusluus) [inline server/routes/urlshortener.js:47]: What's this?

Comment (l8518) [reply server/routes/urlshortener.js:47]: whoops something I should delete :D my bad

File: server/routes/urlshortener.js [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
 const { urlShorten } = require('../models/urlShorten');
 const { ShortenedUrlsRepository } = require('../services/shortenedUrlRepository');
 const { isUri } = require('../helper/web-url-validation');
-const { generate } = require('../middleware/shortid');
 const { IndexNotFoundError, NotFoundError } = require('../models/customErrors');
+const code = require('../helper/code');
 
 var urlRepository = new ShortenedUrlsRepository();
+
+var keys = [];
```
Comment (luusluus) [inline server/routes/urlshortener.js:10]: Does this belong in the controller/route file? I'd rather say in a shortener helper function or something.

**Comment by l8518** [comment]
npm run serve in client folder - or built the static files.

File: server/routes/urlshortener.js (hunk 2 above)
Comment (l8518) [reply server/routes/urlshortener.js:10]: Yes that kinda belongs to the TODOs will move this stuff in another PR

**Comment by luusluus** [comment]
> npm run serve in client folder - or built the static files.

Ok I should install vue-js

File: client/src/components/UrlShortenerClient.vue [hunk 3]
```diff
@@ -0,0 +1,230 @@
+<template>
+  <div class="mt-4">
+    <b-container>
+      <b-row class="mb-4">
+        <b-col cols="7" class="text-left">
+          <b-form @submit="onShortendSubmit" v-if="show">
+            <b-input-group prepend="Your shortend URL" class>
```
Comment (luusluus) [inline client/src/components/UrlShortenerClient.vue:7]: What does shortend mean? Maybe just "Your URL"?

**Pull request merged**